idaes/models_extra/column_models/MEAsolvent_column.py

```diff
@@ -0,0 +1,1483 @@
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info_high(
+            "Step 3 complete: {}.".format(idaeslog.condition(res)))
+
+        # ---------------------------------------------------------------------
+        init_log.info('Step 4a: Isothermal absoption')
+        init_log.info_high("Calculating mass flux")
+
+        # Unfix mass transfer terms
+        blk.interphase_mass_transfer.unfix()
+
+        # Activate mass transfer equation in vapor phase
+        blk.interphase_mass_transfer_eqn.activate()
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+
+        init_log.info('Step 4b: Isothermal chemical absoption')
+        init_log.info_high("Adding mass transfer to material balances")
+
+        blk.vapor_phase.mass_transfer_term.unfix()
+        blk.liquid_phase.mass_transfer_term.unfix()
+        blk.vapor_mass_transfer_eqn.activate()
+        blk.liquid_mass_transfer_eqn.activate()
+
+        # Fix this
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+        init_log.info_high(
+            "Step 4 complete: {}.".format(idaeslog.condition(res)))
+
+        # ---------------------------------------------------------------------
+        init_log.info('Step 5: Adiabatic chemical absoption')
+        init_log.info_high("Isothermal to Adiabatic ")
+
+        # Unfix heat transfer terms
+        blk.heat_transfer_coeff.unfix()
+        for c in ["heat_transfer_Ackmann_correction"]:
+            getattr(blk, c).activate()
+            
+        for k in blk.heat_transfer_Ackmann_correction:
+            calculate_variable_from_constraint(
+                    blk.heat_transfer_coeff[k],
+                    blk.heat_transfer_Ackmann_correction[k])
+        
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+            
+        blk.vapor_phase.heat.unfix()
+        blk.liquid_phase.heat.unfix()
+        blk.vapor_phase.enthalpy_transfer.unfix()
+        blk.liquid_phase.enthalpy_transfer.unfix()
+
+        # Activate heat transfer equations
+        for c in [
+                  "heat_transfer_eqn1",
+                  "heat_transfer_eqn2",
+                  "enthalpy_transfer_eqn1",
+                  "enthalpy_transfer_eqn2"]:
+            getattr(blk, c).activate()
+
+        with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:
+            res = opt.solve(blk, tee=slc.tee)
+
+        init_log.info_high(
+            "Step 5 complete: {}.".format(idaeslog.condition(res)))
+
+        # ---------------------------------------------------------------------
+        blk.vapor_phase.release_state(flags=vflag)
+        blk.liquid_phase.release_state(flags=lflag)
+
+        if (res.solver.termination_condition != TerminationCondition.optimal or
```

Comment:
You should use `check_optimal_termination` from `pyomo.environ` here instead of these two conditions. It does the same thing but is more compact, and also is capable of handling solvers other than IPOPT with different exit codes.